proposed/flexible-http.md

```diff
@@ -0,0 +1,151 @@
+# Flexible HTTP APIs
+
+[Cory Nelson](https://github.com/scalablecory), [Geoff Kizer](https://github.com/geoffkizer)
+
+This design doc describes a set of flexible, high-performance APIs for HTTP.
+
+In shorthand, this may be referred to as "LLHTTP".
+
+## Target customers
+
+This API is intended for advanced users who need control or performance beyond what `HttpClient` offers. `HttpClient` would be kept as the "easy" API, and this API would be the exposed layer that exists beneath `HttpClient`.
```

Comment:
might be worth mentioning that this will only apply to platforms that use SocketsHttpHandler today since on  WebAssembly and in some Xamarin scenarios (where you're using the native platform handler) this won't be possible.

Reply:
Thanks, this is good feedback.